terraform/developer-guide.md

```diff
@@ -0,0 +1,39 @@
+## How to create and setup GCP projects using terraform
+
+### Prerequisites
+
+- Make sure [terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/gcp-get-started) is installed
+- Make sure you have access to `who-terraform-admin` project.
+- Create a new key for the `terraform@who-terraform-admin.iam.gserviceaccount.com` service account in the `who-terraform-admin` project, download the JSON credentials, and configure your environment:
+
+  ```sh
+  export GOOGLE_APPLICATION_CREDENTIALS=<path to json credentials>
+  ```
+
+- Have a GCP billing account:
+
+  ```sh
+  export TF_VAR_billing_account=<GCP billing account id>
+  ```
+
+### Create or update a GCP project
+
+- Configure environment:
+
+  ```sh
+  export TF_VAR_project_name=<project name>
```

Comment:
I think the instances should be specified in tf-var files. This puts it under source control: https://www.terraform.io/docs/configuration/variables.html#variable-definitions-tfvars-files

So the documented approach ends up with distinct files for prod, staging, hack, dev and so on. Staging and hack should be automatic pushes. Prod and dev can be manually done. Not sure if you removed `=` then the autocomplete would work which would be nice.

I like this better as the action is more obvious and not hidden by environment variables:
```
terraform apply -var-file=staging.tfvars
```